v1/ao/opentelemetry/tracer.go

```diff
@@ -0,0 +1,117 @@
+package opentelemetry
+
+import (
+	"bytes"
+	"context"
+	"strings"
+
+	"github.com/appoptics/appoptics-apm-go/v1/ao"
+	"github.com/appoptics/appoptics-apm-go/v1/ao/internal/reporter"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// tracerImpl implements the Tracer interface of OpenTelemetry
+type tracerImpl struct {
+	name string
+}
+
+var _ trace.Tracer = &tracerImpl{}
+
+// Start a span
+func (t *tracerImpl) Start(ctx context.Context, name string, options ...trace.SpanOption) (context.Context, trace.Span) {
+	config := trace.NewSpanConfig(options...)
+
+	parentSpan, links := GetSpanContextAndLinks(ctx, config.NewRoot)
+	s := &spanImpl{tracer: t, name: name, spanKind: config.SpanKind}
+	if parentSpan != nil {
+		s.parent = parentSpan // TODO we don't seem to need it?
+		parentAoSpan := parentSpan.(*spanImpl)
+		s.aoSpan = parentAoSpan.aoSpan.BeginSpanWithOptions(s.name, ao.SpanOptions{
+			StartTime: config.Timestamp,
+		})
+	} else {
+		mdStr := ""
+		s.aoSpan = ao.NewTraceWithOptions(s.name, ao.SpanOptions{
+			StartTime: config.Timestamp,
+			ContextOptions: ao.ContextOptions{
+				MdStr: mdStr,
+			},
+		})
+	}
+
+	s.context = MdStr2OTSpanContext(s.aoSpan.MetadataString())
+	ctx = trace.ContextWithSpan(ctx, s)
+
+	for _, l := range links {
+		s.addLink(l)
+	}
+	for _, l := range config.Links {
+		s.addLink(l)
+	}
+	s.SetAttributes(config.Attributes...)
+
+	return ctx, s
+}
+
+// GetSpanContextAndLinks get local and remote SpanContext from the context, and
+// produces the links accordingly.
+// This is based from the OpenTelemetry sdk code.
+func GetSpanContextAndLinks(ctx context.Context,
+	ignoreContext bool) (trace.Span, []trace.Link) {
+	local := trace.SpanFromContext(ctx)
+
+	if ignoreContext {
+		links := addLinkIfValid(nil, local.SpanContext(), "current")
```

Comment:
Is this trying to keep some info of current context (if exist), even if the config tells creating a new root? :)

Reply:
Yes, this produces a loose link to the current span context with an attribute indicating it's "ignored-on-demand".